### PR TITLE
Adding tests to #506

### PR DESCRIPTION
**Pull request by jbusecke**
I made some test that trigger the bug exposed in #504, and thought these might be helpful here.

### CODE REVIEW

File: tests/test_end_to_end.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def pipeline():
         yield p
 
 
-@pytest.mark.parametrize("target_chunks", [{"time": 1}, {"time": 2}, {"time": 3}])
+@pytest.mark.parametrize("target_chunks", [{"time": 1}, {"time": 2}, {"time": 3}, {'time':1, 'lon': 18}])
```
Comment (jbusecke) [inline tests/test_end_to_end.py:22]: Ensures that providing target chunks to additional dimension which are part of the data variable does not trigger the failure

**Comment by rabernat** [comment]
Thanks so much for this Julius! I really appreciate your help with adding tests for this bug.

What you have added here is an end-to-end test, which is quite expensive and potentially also more fragile. I would like to see if we can cover this with a more narrowly scoped test, specifically here

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/661d13b692492ff5b127f81a6d3d3dd3a3c63798/tests/test_aggregation.py#L21-L37

Overall, one motivation for the beam refactor was to move away from these massive end-to-end tests towards more focused tests. We can't hope to add an end-to-end test for every possible data edge case out there. 

So, if you have time, I would like you to retry this with a unit test in `test_aggregation.py` instead of an end-to-end test. This will force you to confront some internals like Schema which will help you become a more effective contributor to this package.

**Comment by rabernat** [comment]
Just pinging on this. Any chance you would be up for refactoring your test? Happy to help with this.

**Comment by jbusecke** [comment]
Def up for it, but not seeing that I will get to it this week unfortunately.

**Comment by cisaacstern** [comment]
The requested unit tests were added in #4.

This can be closed.

Thanks @jbusecke!

**Comment by jbusecke** [comment]
Closing this
